src/stream/gzip.rs

```diff
@@ -42,15 +42,15 @@ impl<S: Stream<Item = Result<Bytes>>> Stream for GzipStream<S> {
             if *this.flushing {
                 if !*this.footer_appended {
                     let mut footer = Bytes::from(&this.crc.sum().to_le_bytes()[..]);
-                    let length_read = &this.crc.amount().to_le_bytes()[..];
-                    footer.extend_from_slice(length_read);
+                    let bytes_read = &this.crc.amount().to_le_bytes()[..];
+                    footer.extend_from_slice(bytes_read);
                     *this.footer_appended = true;
                     return Poll::Ready(Some(Ok(footer)));
                 } else {
                     return Poll::Ready(None);
                 }
             } else if let Some(bytes) = ready!(this.inner.poll_next(cx)) {
-                *this.input_buffer = bytes?;
+                this.input_buffer.extend_from_slice(&bytes?);
```

Comment:
Using `extend_from_slice` here will cause an unnecessary copy.

Reply:
Hmm, yeah... I was in the mindset that I would always be keeping the last 8 bytes so it would make it a necessity to do that but I might not have to.